### ISSUE TITLE
promisify(fse.ensureDir) in examples doesn't work as expected in case of nested directories

### ISSUE THREAD

**Issue by azotova** [nodegit/nodegit#996]
I've tried following the `add-and-commit` example and noticed that I get an ENOENT error when `fse.ensureDir` has to create nested directories. Using it without the promise (and putting the rest of the code in the callback of `fse.ensureDir`) solved the issue.

**Comment by tbranyen** [comment]
Can you elaborate on how you got the example to break? I just ran the code on my Arch Linux machine with the master branch snapshot and it worked fine. There is a nested path in the example so that should confirm the behavior you're describing, but I'm not seeing it.

Maybe it's OS related? More details please :ok_hand:

**Comment by azotova** [comment]
Yes, I ran it on OS. The important thing is that in my example `ensureDir` had to create both the parent and the child directory. I save files with a path that looks something like this `/fullpath/drafts/folder1/folder2/file1.pug`. I followed the `add-and-commit` example pretty closely. The line from the example `return fse.ensureDir(path.join(repo.workdir(), directoryName));` in my case looked like this: `fse.ensureDir(path.dirname(fileName))`. If `folder1` existed and `folder2` had to be created, everything worked fine. However, if both `folder1` and `folder2` had to be created, I got ENOENT error. I was able to reproduce it consistently.

**Comment by azotova** [comment]
I tested it on Mac and it worked fine. So I experience this issue only in the Docker container. The image in the Docker container is based on Debian GNU/Linux 8 (jessie). Sorry for not testing properly before reporting.

**Comment by saper** [comment]
Some read-only filesystem somewhere?
